main.go

```diff
@@ -789,31 +789,57 @@ func (s *Server) RestartAllTransfersForLocation(ctx context.Context, loc string)
 			continue
 		}
 
-		if err := s.CM.RestartTransfer(ctx, loc, chid, d.ID); err != nil {
+		if err := s.CM.RestartTransfer(ctx, loc, chid, d); err != nil {
 			log.Errorf("failed to restart transfer: %s", err)
 			continue
 		}
 	}
 	return nil
 }
 
-func (cm *ContentManager) RestartTransfer(ctx context.Context, loc string, chanid datatransfer.ChannelID, dealID uint) error {
-	if loc == "local" {
+func (cm *ContentManager) RestartTransfer(ctx context.Context, loc string, chanid datatransfer.ChannelID, d contentDeal) error {
+	maddr, err := d.MinerAddr()
+	if err != nil {
+		return err
+	}
+
+	var dealUUID *uuid.UUID
+	if d.DealUUID != "" {
+		parsed, parseErr := uuid.Parse(d.DealUUID)
```

Comment:
I think we might want to error out if `d.DealUUID != nil` but there's a `parseErr`. Because we assume the user wants us to use that UUID but we can't understand it.